velox/type/Type.h

```diff
@@ -1521,10 +1526,28 @@ inline ComplexType to(std::string value) {
   return ComplexType();
 }
 
+/// Forward declaration.
+namespace exec {
+
+class CastOperator;
+
+} // namespace exec
+
+using CastOperatorPtr = std::shared_ptr<exec::CastOperator>;
```

Comment:
since CastOperator is in exec namespace, I think it makes sense to put CastOperatorPtr into exec namespace as well.

Reply:
Does it also make sense to have JsonCastOperator in `velox::exec` namespace? This will make JsonType.h contain some contents in `velox` namespace and others in `velox::exec` though.

Reply:
I don't think so. JsonType is Presto-specific, hence, belongs in xxx::prestosql namespace.

Reply:
Thank you for the reply! I found that `TimestampWithTimeZoneType` and `HyperLogLogType` are in the `facebook::velox` namespace, so I'm keeping `JsonType` in the same one. Please let me know if you have other suggestions.